### PR TITLE
Find Similar Colors

### PR DESCRIPTION
**Pull request by terraoldham**
- Reloading "find similar colors" data into the library
- Adding ability to sort by name, color, and price
- Updating tint color on action sheets

/review @bblia @alexritchey

### CODE REVIEW

File: nailed-it/New Group/PolishLibraryViewController.swift [hunk 1]
```diff
@@ -193,55 +242,111 @@ extension PolishLibraryViewController: CZPickerViewDelegate, CZPickerViewDataSou
...
 
-    func czpickerView(_ pickerView: CZPickerView!, didConfirmWithItemAtRow row: Int){
-        self.navigationController?.setNavigationBarHidden(true, animated: true)
     }
 
     func czpickerViewDidClickCancelButton(_ pickerView: CZPickerView!) {
         self.navigationController?.setNavigationBarHidden(false, animated: true)
     }
+    
+    func czpickerView(_ pickerView: CZPickerView!, didConfirmWithItemAtRow row: Int){
```
Comment (DexOfTheWild) [inline nailed-it/New Group/PolishLibraryViewController.swift]: Can you give me a quick rundown of what this is doing?

Comment (terraoldham) [reply nailed-it/New Group/PolishLibraryViewController.swift]: Yep, this is a pop up view that allows users to sort/filter. It's from the CZPicker cocoapod. 

screen shot 2017-10-22 at 3 03 01 pm
screen shot 2017-10-22 at 3 02 55 pm

File: nailed-it/Models/LibraryVector.swift [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+//  LibraryVector.swift
+//  nailed-it
+//
+//  Created by Terra Oldham on 10/20/17.
+//  Copyright © 2017 Lia Zadoyan. All rights reserved.
+//
+
+import UIKit
+
+class LibraryVector: NSObject {
```
Comment (bblia) [inline nailed-it/Models/LibraryVector.swift]: do we need this class if its empty?

File: nailed-it/New Group/PolishLibraryViewController.swift [hunk 3]
```diff
@@ -42,21 +45,37 @@ class PolishLibraryViewController: UIViewController, UICollectionViewDelegate, U
...
+        if type == "sort" {
+            picker.allowMultipleSelection = false
+            self.sortFilter = "sort"
+        } else {
+            picker.allowMultipleSelection = true
+            self.sortFilter = "filter"
+        }
+        picker.headerBackgroundColor = greenColor
+        picker.confirmButtonBackgroundColor = greenColor
+        picker.headerTitleColor = pinkColor
```
Comment (bblia) [inline nailed-it/New Group/PolishLibraryViewController.swift]: nitpick, but can we make the font colors white? like the title font and the button fonts. its just easier to see on a colored background

File: nailed-it/New Group/PolishLibraryViewController.swift [hunk 4]
```diff
@@ -189,59 +238,117 @@ extension UIImage {
...
+        if self.sortFilter == "sort" {
+            if self.sortingOptions[row] == "Price: $ to $$$" {
+                startAnimating(size, message: "Sorting...", type: NVActivityIndicatorType.ballTrianglePath)
+                // Sorting based on brand right now, because it corresponds to price
+                let sortedColors = self.colors?.sorted {
+                    let string0 = String(describing: $0.brand)
+                    let string1 = String(describing: $1.brand)
+                    return string0 > string1
+                }
+                self.stopAnimating()
```
Comment (bblia) [inline nailed-it/New Group/PolishLibraryViewController.swift]: these next 3 lines are repeated in almost every block, can we consolidate into a function?

File: nailed-it/New Group/PolishLibraryViewController.swift [hunk 5]
```diff
@@ -158,7 +196,36 @@ class PolishLibraryViewController: UIViewController, UICollectionViewDelegate, U
...
+            let blueDistance = color.blueValue - (libraryColor?.blueValue)!
+            libraryColor?.distanceVector = CGFloat(((redDistance * redDistance) + (greenDistance * greenDistance) + (blueDistance * blueDistance)).squareRoot())
+            
+        }
+        let sortedColors = self.colors?.sorted {
+            let string0 = String(describing: $0.distanceVector)
+            let string1 = String(describing: $1.distanceVector)
+            return string0 < string1
+        }
+        self.stopAnimating()
```
Comment (bblia) [inline nailed-it/New Group/PolishLibraryViewController.swift]: we should stop animating after the reload and scrollToItem

File: nailed-it/New Group/PolishLibraryViewController.swift [hunk 6]
```diff
@@ -158,7 +196,36 @@ class PolishLibraryViewController: UIViewController, UICollectionViewDelegate, U
...
+        self.collectionView.scrollToItem(at: IndexPath(row: 0, section: 0),
+                                          at: .top,
+                                          animated: true)
 
+    }
+    
+    func updateSortedColors(sortedColors: [PolishColor]) {
+        self.stopAnimating()
+        self.colors = sortedColors
+        self.collectionView.reloadData()
```
Comment (bblia) [inline nailed-it/New Group/PolishLibraryViewController.swift]: same here with the animating, after the reload i think

**Comment by bblia** [comment]
lgtm

**Comment by bblia** [comment]
lgtm

**Pull request merged**